core/blockchain.go

```diff
@@ -143,6 +143,55 @@ var defaultCacheConfig = &CacheConfig{
 	SnapshotWait:   true,
 }
 
+type BlockChainConfig struct {
+	CacheConfig    *CacheConfig
+	ChainConfig    *params.ChainConfig
+	VmConfig       vm.Config
+	ShouldPreserve func(*types.Header) bool
+	TxLookupLimit  *uint64
+}
+type BcConfigOpt func(*BlockChainConfig)
+
+func SetCacheConfig(cacheConfig *CacheConfig) BcConfigOpt {
+	return func(bcConfig *BlockChainConfig) {
+		bcConfig.CacheConfig = cacheConfig
+	}
+}
+func SetChainConfig(chainConfig *params.ChainConfig) BcConfigOpt {
```

Comment:
Exported functions like this should have comments explaining them

Reply:
Yep  thanks for the suggestion and I've added comment lines explaining each of the functions added.